lib/tasks/db.rake

```diff
@@ -1,10 +1,49 @@
 namespace :db do
+  def db_allow_task?
+    if Rails.env.test? || Rails.env.development?
+      true
+    else
+      puts "Only allowed in development and test mode"
+      false
+    end
+  end
 
   desc "Drop database for current RAILS_ENV"
   task :oracle_drop => :environment do
+    next unless db_allow_task?
     config= Rails.configuration.database_configuration[Rails.env]
     connect_string = "#{config["username"]}/#{config["password"]}@#{config["database"]}"
     system "bash -ic 'cd #{Rails.root}/db && ./generate_drops.sh | sqlplus #{connect_string}'"
```

Comment:
I was thinking more like:

```ruby
Dir.chdir(Rails.root.join("db")) do
  system "./generate_drops.sh | sqlplus #{Shellwords.shellescape(connect_string)}"
end
```

I'm not sure the `bash -ic` needs to be there, since I would expect the environment to be set up correctly already when running this from the command line.